### PR TITLE
base fix for #54, add print button for backup codes

### PR DESCRIPTION
**Pull request by voldemortensen**

### CODE REVIEW

File: providers/class.two-factor-backup-codes.php [hunk 1]
```diff
@@ -140,7 +141,7 @@ public function user_options( $user ) {
 						dataType: 'JSON',
 						success: function( response ) {
 							$( '.two-factor-backup-codes-wrapper' ).show();
-							$( '.two-factor-backup-codes-unused-codes' ).html( '' );
+							$( '.two-factor-backup-codes-unused-codes' ).html( '<p>Backup codes for ' + document.domain + '</p>' );
```
Comment (valendesigns) [inline providers/class.two-factor-backup-codes.php]: Text should be translatable.

File: user-edit.css [hunk 2]
```diff
@@ -34,4 +34,24 @@
...
 .two-factor-methods-table tbody tr:nth-child(odd) {
 	background-color: #f9f9f9;
-}
\ No newline at end of file
+}
+
+@media print {
+	body * {
+		visibility: hidden;
+		max-height: 100px;
```
Comment (valendesigns) [inline user-edit.css]: What is the `max-height` for?

Comment (voldemortensen) [inline user-edit.css]: This was an attempt to make sure it didn't print 2-3 extra blank pages. It didn't work however. In my print preview it still shows 2 extra pages.

Comment (valendesigns) [inline user-edit.css]: Try using display instead of visibility.

Comment (voldemortensen) [inline user-edit.css]: display doesn't work because it hides all children, and you can't use display to get children back if the parent isn't shown.

Comment (valendesigns) [inline user-edit.css]: Possibly a hidden overflow could work with max-height, or simply use an explicit height and absolute positioning on all elements?

**Comment by valendesigns** [comment]
Since we're relying on JS to print the backup codes, I feel we should go all-in and not relying on CSS to hide and show everything else. I couldn't find a reliable way to display a clean looking UI without a second page being added to the print queue. Instead I opened a new window to use as our print document and wrote it out manually. Let me know if any of you have objections or concerns with this approach.

**Comment by voldemortensen** [comment]
I think we should do that as a no js fallback. I could be wrong, but I think opening a new window goes against accessibility. I'd have to have someone versed in that to confirm though.

**Comment by valendesigns** [comment]
There could be accessibility concerns we need to address, but I'm not aware of any personally. We'll need a third party to weigh in on the subject. However, my concerns with a CSS approach are simple. We can't assume that 2fa backup codes is the only reason someone would have to print the user profile and the method for hiding other elements is buggy at best. If we can address those concerns with a CSS only approach then I'm all for it. Otherwise, we need to do what works. I'm not sure there is an ideal approach to this issue, just different ones with trade offs.

**Comment by stevenkword** [comment]
Can we elaborate on the new window / accessibility argument?  That's how Google is doing it.

https://accounts.google.com/b/0/SmsAuthSettings

**Comment by stevenkword** [comment]
We should also consider the way GitHub handles this process. They use a button, but when you click it, there is no print / modal. It is a "download & print”, so it never leaves the page and prompts you to save the TXT file locally.

https://github.com/settings/security

**Pull request closed without merging**